src/app/features/ledger/hooks/use-ledger-navigate.ts

```diff
@@ -42,7 +43,14 @@ export function useLedgerNavigate() {
             state: { tx: bytesToHex(psbt) },
           });
         }
-
+        const requestToken = searchParams.get('request');
+        if (requestToken) {
+          return navigate(RouteUrls.ConnectLedger + `?request=${requestToken}`, {
+            replace: true,
+            relative: 'route',
+            state: { tx: bytesToHex(psbt) },
+          });
+        }
```

Comment:
I don't think it's necessary to do any special handling here because request token is already persisted in initialSearchParams